selectors/views/threads.ts

```diff
@@ -72,3 +91,87 @@ export const isThreadOpen = (state: GlobalState, threadId: $ID<UserThread>): boo
 export const isThreadManuallyUnread = (state: GlobalState, threadId: $ID<UserThread>): boolean => {
     return state.views.threads.manuallyUnread[threadId] || false;
 };
+
+// Returns a selector that, given the state and an object containing an array of postIds and an optional
+// timestamp of when the channel was last read, returns a memoized array of postIds interspersed with
+// day indicators and an optional new message indicator.
+export function makeFilterRepliesAndAddSeparators() {
+    const getPostsForIds = makeGetPostsForIds();
+
+    return createIdsSelector(
+        'makeFilterPostsAndAddSeparators',
+        (state: GlobalState, {postIds}: PostFilterOptions) => getPostsForIds(state, postIds),
+        (_state: GlobalState, {lastViewedAt}: PostFilterOptions) => lastViewedAt,
+        (_state: GlobalState, {showDate}: PostFilterOptions) => showDate,
+        (_state: GlobalState, {collapsedThreads}: PostFilterOptions) => collapsedThreads,
+        (_state: GlobalState, {openTime}: PostFilterOptions) => openTime,
```

Comment:
What is `openTime` used for? I can see below that it does something with ephemeral posts, but I'm not quite sure what.

Reply:
The case is to not show ephemeral posts that have been received before opening the ThreadViewer.

Reply:
Is that a new use case or is that from the old RHS code? That sounds rare enough that I'd consider just dropping it to help simplify this code a bit.

Reply:
it's from old RHS code, so we won't have an ephemeral post shown both in center and RHS

Reply:
Oh yeah, I remember that logic now. It was always a rather weird case. I'm always glad to get rid of odd cases like that